### ISSUE TITLE
Version 2.1.0 breaks on Apache

### ISSUE THREAD

**Issue by Y0lk** [zendframework/zend-diactoros#346]
- [x] I was not able to find an [open](https://github.com/zendframework/zend-diactoros/issues?q=is%3Aopen) or [closed](https://github.com/zendframework/zend-diactoros/issues?q=is%3Aclosed) issue matching what I'm seeing.
 - [x] This is not a question. (Questions should be asked on [chat](https://zendframework.slack.com/) ([Signup here](https://zendframework-slack.herokuapp.com/)) or our [forums](https://discourse.zendframework.com/c/questions/expressive).)

Since the update to 2.1.0, I'm getting the following error on a local Apache install on Windows. I haven't tested this anywhere else yet.

> Fatal error: Uncaught Zend\Diactoros\Exception\InvalidArgumentException: "<address>Apache/2.4.9 (Win32) PHP/7.1.24 Server at artofwhere.local Port 80</address> " is not valid header value

The exact change that causes this issue is this commit: https://github.com/zendframework/zend-diactoros/commit/1e8b6c4db2aed9647945f86a3e46178fca509e04#diff-9d47206c200ee53ffe55d05493ddd000

I've tried commenting out the line that was added and it fixes the issue

### Code to reproduce the issue

```php
$request = \Zend\Diactoros\ServerRequestFactory::fromGlobals(
    $_SERVER,
    $_GET,
    $_POST,
    $_COOKIE,
    $_FILES
);
```

**Comment by bluebaroncanada** [comment]
@Y0lk, you can fix this for now by deleting line 50 from `marshal_headers_from_sapi.php`.

**Comment by Y0lk** [comment]
Yeah, as I wrote in the description, it's the commit that added this line 50 that causes problem. You can just force the version to 2.0.1 in your composer.json for a cleaner fix for now

**Comment by bluebaroncanada** [comment]
Ah.  Sorry.  I actually came upon this issue after I made the pull request. Sorry I didn't read it well enough.  He was trying to fix an error where he was assuming headers with a 0 value were not being accepted.  He ended up defeating the filter in the process. I think he just needs to prefix his headers with `HTTP_`.

**Comment by bluebaroncanada** [comment]
#349
